api/open_saves.proto

```diff
@@ -89,6 +89,76 @@ service OpenSaves {
+  // For example, CompareAndSwapProperty(property, value = 42, old_value = 24) will set the
+  // property to 42 if the current value is 24.
+  // CompareAndSwapProperty also supports swapping with a value of another type, e.g. 
+  // CompareAndSwapProperty(property, value = "42", old_value = 24).
+  // Otherwise it will not update the property and return the current (unchanged) value
+  // and updated = false. The operation is executed atomically.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  rpc CompareAndSwapProperty(CASPropertyRequest) returns (CASPropertyResponse) {}
+
+  // CompareAndSwapGreaterInt compares the number of an integer property to value and
+  // updates the property if the new value is greater than the current value.
+  // The updated field in AtomicResponse is set to true if the swap is executed.
+  // For example, CompareAndSwapGreaterInt(property, value = 42) will replace property with 42
+  // and return {value = old value, updated = true} if 42 > property.
+  // Otherwise it will not update the property and return the current (unchanged) value
+  // and updated = false. The operation is executed atomically.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  //   - InvalidArgument: the requested property was not an integer.
+  rpc CompareAndSwapGreaterInt(AtomicIntRequest) returns (AtomicIntResponse) {}
+
+  // CompareAndSwapLessInt does the same operation as CompareAndSwapGreaterInt except
+  // the condition is the new value is less than the old value.
+  rpc CompareAndSwapLessInt(AtomicIntRequest) returns (AtomicIntResponse) {}
+
+  // AtomicAddInt adds a number to an integer property atomically.
+  // For example, AtomicAdd(property, 42) will run property += 42 and return the old value.
+  // The updated field in AtomicIntResponse is always set to true.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  //   - InvalidArgument: the requested property was not an integer.
+  rpc AtomicAddInt(AtomicIntRequest) returns (AtomicIntResponse) {}
+
+  // AtomicSubInt does the same except it subtracts a number.
+  rpc AtomicSubInt(AtomicIntRequest) returns (AtomicIntResponse) {}
+
+  // AtomicIncInt increments the number of an integer property if less than upper_bound. 
+  // Otherwise it resets the property to lower_bound.
+  // if (property < upper_bound) {
+  //   property++
+  // } else {
+  //   property = lower_bound
+  // }
+  // This makes the property an incrementing counter between [lower_bound, upper_bound].
+  // It returns the old value of the property.
+  // The updated field in AtomicIntResponse is set to true.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  //   - InvalidArgument: the requested property was not an integer.
+  rpc AtomicIncInt(AtomicIncIntRequest) returns (AtomicIntResponse) {}
```

Comment:
This is probably the hardest to read and remember.  I would simply call the functions:
AtomicIncrement or AtomicInc
AtomicDecrement or AtomicDec